server/src/main/java/org/opensearch/search/aggregations/bucket/composite/CompositeAggregationFactory.java

```diff
@@ -80,7 +81,7 @@ protected Aggregator createInternal(
 
     @Override
     protected boolean supportsConcurrentSegmentSearch() {
-        // See https://github.com/opensearch-project/OpenSearch/issues/12331 for details
-        return false;
+        // Disable concurrent search if any scripting is used. See https://github.com/opensearch-project/OpenSearch/issues/12331 for details
+        return Arrays.stream(sources).allMatch(CompositeValuesSourceConfig::hasScript) == false;
```

Comment:
Should this be `anyMatch`?

Reply:
Yeah I think I got this one mixed up in my head -- it should really just be `noneMatch` which should be equivalent to `anyMatch == false`